go.mod

```diff
@@ -17,17 +21,22 @@ require (
 	k8s.io/controller-manager v0.33.0
 	k8s.io/klog/v2 v2.130.1
 	k8s.io/utils v0.0.0-20250502105355-0f33e8f1c979
+	sigs.k8s.io/gateway-api v1.3.0
```

Comment:
it would be really nice if we could import just the API without bringing dependencies like DNS, controller-runtime, etc into our dependency graph